File: app/src/main/java/com/cqebd/student/live/custom/CustomAttachParser.java
package com.cqebd.student.live.custom;

import com.netease.nimlib.sdk.msg.attachment.MsgAttachment;
import com.netease.nimlib.sdk.msg.attachment.MsgAttachmentParser;

/**
 * Created by xiaofu on 2018/5/30.
 */
public class CustomAttachParser implements MsgAttachmentParser {

    @Override
    public MsgAttachment parse(String json) {
        CustomAttachment attachment = new NormalAttachment();
        attachment.fromJson(json);
        return attachment;
    }
}

File: lib_netease/src/main/java/com/cqebd/lib_netease/customMsg/CustomAttachParser.kt
package com.cqebd.lib_netease.customMsg

import com.netease.nimlib.sdk.msg.attachment.MsgAttachment
import com.netease.nimlib.sdk.msg.attachment.MsgAttachmentParser

class CustomAttachParser : MsgAttachmentParser {

    override fun parse(json: String): MsgAttachment {
        lateinit var attachment: CustomAttachment
//        var attachment:CustomAttachment? = null

        attachment.fromJson(json)

        return attachment
    }
}